kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/convert/BigQueryRecordConverter.java

```diff
@@ -73,13 +66,60 @@ public BigQueryRecordConverter(boolean shouldConvertDoubleSpecial) {
    */
   public Map<String, Object> convertRecord(SinkRecord kafkaConnectRecord) {
     Schema kafkaConnectSchema = kafkaConnectRecord.valueSchema();
+    Object kafkaConnectValue = kafkaConnectRecord.value();
+    if (kafkaConnectSchema == null) {
+      if (kafkaConnectValue instanceof Map) {
+        return (Map<String, Object>) convertSchemalessRecord(kafkaConnectValue);
+      }
+      throw new ConversionConnectException("Only Map objects supported in absence of schema for " +
+              "record conversion to BigQuery format.");
+    }
     if (kafkaConnectSchema.type() != Schema.Type.STRUCT) {
       throw new
           ConversionConnectException("Top-level Kafka Connect schema must be of type 'struct'");
     }
     return convertStruct(kafkaConnectRecord.value(), kafkaConnectSchema);
   }
 
+  private Object convertSchemalessRecord(Object value) {
+    if (value == null) {
+      return null;
+    }
+    if (value instanceof Double) {
+      return convertDouble((Double) value);
+    }
+    if (ClassUtils.isPrimitiveWrapper(value.getClass()) || value instanceof String) {
+      return value;
+    }
+    if (value instanceof byte[] || value instanceof ByteBuffer) {
+      return convertBytes(value);
+    }
+    if (value instanceof List) {
+      return
+          ((List) value).stream().map(
+                  v -> convertSchemalessRecord(v)
+          ).collect(Collectors.toList());
+    }
+    if (value instanceof Map) {
+      return
+        ((Map<Object, Object>) value).entrySet().stream().collect(
+                Collectors.toMap(
+                        entry -> {
+                          if (!(entry.getKey() instanceof String)) {
+                            throw new ConversionConnectException(
+                                    "Map objects in absence of schema needs to have string value keys. " +
```

Comment:
I feel like the ordering of these messages should be reversed:
"Failed to convert schemaless record to BigQuery format: map objects need string value keys"